interp/constrintern.ml

```diff
@@ -1634,7 +1624,8 @@ let merge_aliases aliases {loc;v=na} =
   match na with
   | Anonymous -> aliases
   | Name id ->
-  let alias_ids = aliases.alias_ids @ [make ?loc id] in
+  let alias_ids = List.add_set (fun id1 id2 -> Id.equal id1.v id2.v) (make ?loc id) aliases.alias_ids in
+  (* Note: assumes that add_set adds in front, see alias_of *)
```

Comment:
As a user I guess I would prefer a warning when I type "foobar as id as id" as it seems more likely to be a typo but why not.

Reply:
A warning would be easy to add. As you want.

Actually, we could even syntactically forbid nesting `as` because it is anyway useless.

Reply:
> Actually, we could even syntactically forbid nesting as because it is anyway useless.

Indeed, that sounds like the best solution.

Reply:
I merged this but we can syntactically forbid nesting as in a further PR.